pkg/processors/frontmatter.go

```diff
@@ -75,3 +83,31 @@ func (f *FrontMatter) Process(documentBlob []byte, node *api.Node) ([]byte, erro
 	}
 	return documentBlob, nil
 }
+
+// Determines node title from its name or its parent name if
+// it is eligible to be index file, and then normalizes either
+// as a title - removing `-`, `_`, `.md` and converting to title
+// case.
+func (f *FrontMatter) getNodeTitle(node *api.Node) string {
+	title := node.Name
+	if node.Parent() != nil && f.nodeIsIndexFile(node.Name) {
+		title = node.Parent().Name
+	}
+	title = strings.TrimRight(title, ".md")
+	title = strings.ReplaceAll(title, "_", " ")
+	title = strings.ReplaceAll(title, "-", " ")
+	title = strings.Title(title)
+	return title
+}
+
+// Compares a node name to the configured list of index file
+// and a default name '_index.md' to determin if this node
+// is an index document node.
+func (f *FrontMatter) nodeIsIndexFile(name string) bool {
```

Comment:
I believe this code is duplication of 
pkg/hugo/processor.go > (*Processor).nodeIsIndexFile